core/tests/test_storage.cache.py

```diff
@@ -44,38 +43,32 @@ def call_Initialize(**kwargs):
             return await_result(handle_Initialize(DUMMY_CONTEXT, msg))
 
         # calling Initialize without an ID allocates a new one
-        session_id = cache.get_session_id()
+        session_id = cache.start_session()
         features = call_Initialize()
-        new_session_id = cache.get_session_id()
-        self.assertNotEqual(session_id, new_session_id)
-        self.assertEqual(new_session_id, features.session_id)
+        self.assertNotEqual(session_id, features.session_id)
 
         # calling Initialize with the current ID does not allocate a new one
-        features = call_Initialize(session_id=new_session_id)
-        same_session_id = cache.get_session_id()
-        self.assertEqual(new_session_id, same_session_id)
-        self.assertEqual(same_session_id, features.session_id)
-
-        call_Initialize()
-        # calling Initialize with a non-current ID returns a different one
-        features = call_Initialize(session_id=new_session_id)
-        self.assertNotEqual(new_session_id, features.session_id)
+        features = call_Initialize(session_id=session_id)
+        self.assertEqual(session_id, features.session_id)
 
-        # allocating a new session ID clears the cache
+        # store "hello"
         cache.set(KEY, "hello")
+        # check that it is cleared
         features = call_Initialize()
+        session_id = features.session_id
         self.assertIsNone(cache.get(KEY))
-
-        # resuming a session does not clear the cache
+        # store "hello" again
         cache.set(KEY, "hello")
-        call_Initialize(session_id=features.session_id)
         self.assertEqual(cache.get(KEY), "hello")
 
-        # supplying a different session ID clears the cache
-        self.assertNotEqual(new_session_id, features.session_id)
-        call_Initialize(session_id=new_session_id)
+        # supplying a different session ID starts a new cache
+        call_Initialize(session_id=b"A")
         self.assertIsNone(cache.get(KEY))
 
+        # but resuming a session loads the previous one
+        call_Initialize(session_id=session_id)
+        self.assertEqual(cache.get(KEY), "hello")
```

Comment:
there should also be a test for two different values stored in two different sessions

Reply:
and a test that when sessions are exhausted, you can't get back the old item from the old session id. this is kinda-sorta subsumed in the above session id test, but i'd still like to have it separate